bigflow/deploy.py

```diff
@@ -35,18 +38,76 @@ def tag_image(image_id: str, repository: str, tag: str) -> str:
 
 
 def deploy_docker_image(
-        image_tar_path: str,
-        docker_repository: str,
-        auth_method: AuthorizationType = AuthorizationType.LOCAL_ACCOUNT,
-        vault_endpoint: T.Optional[str] = None,
-        vault_secret: T.Optional[str] = None,
+    image_tar_path: str,
+    docker_repository: str,
+    auth_method: AuthorizationType = AuthorizationType.LOCAL_ACCOUNT,
+    vault_endpoint: T.Optional[str] = None,
+    vault_secret: T.Optional[str] = None,
 ) -> str:
+    if image_tar_path.endswith(".toml"):
+        deploy_fn = deploy_docker_image_from_local_repo
+    else:
+        deploy_fn = deploy_docker_image_from_fs
+    deploy_fn(
+        image_tar_path,
+        docker_repository,
+        auth_method,
+        vault_endpoint,
+        vault_secret,
+    )
+
+
+def deploy_docker_image_from_local_repo(
```

Comment:
this method is very similar to `_deploy_image_loaded_to_local_registry` can not we scale them?

Reply:
Refactored.